server/webapp/WEB-INF/rails.new/app/assets/new_javascripts/views/auth_configs/auth_config_modal_body.js.msx

```diff
@@ -0,0 +1,174 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+define([
+  "mithril", 'lodash', 'jquery', 'helpers/form_helper', 'models/pipeline_configs/plugin_infos', 'models/auth_configs/auth_configs',
+  'views/shared/angular_plugin'
+], function (m, _, $, f, PluginInfos, AuthConfigs, AngularPlugin) {
+
+  var autoComplete = function (model, attrName, selectCb) {
+    return function (elem, isInitialized, context) {
+      if (isInitialized) {
+        return;
+      }
+
+      var $inputElem = $($(elem).find("input[data-prop-name='" + attrName + "']"));
+      var values     = _.map(PluginInfos(), function (pi) {
+        return pi.id();
+      });
+
+      context.onunload = function () {
+        $inputElem.textcomplete('destroy');
+      };
+
+      $inputElem.textcomplete([
+        {
+          match:   /([^\s].*)$/,
+          index:   1, // use the second match in the regex to extract the search term
+          search:  function (term, callback) {
+            term = term.toLowerCase();
+            callback($.map(values, function (word) {
+              return word.toLowerCase().indexOf(term) >= 0 ? word : null;
+            }));
+          },
+          replace: function (word) {
+            return word;
+          }
+        }
+      ], {
+        zIndex: '10000'
+      });
+
+      $inputElem.on('textComplete:select', function () {
+        model()[attrName]($inputElem.val());
+        if (selectCb) {
+          selectCb($inputElem.val());
+        }
+      });
+    };
+  };
+
+  var AuthConfigModalBody = {
+    controller: function (args) {
+      var ctrl          = this;
+      var setPluginInfo = function (pluginInfo) {
+        args.pluginInfo(pluginInfo);
+        args.saveDisabled(!pluginInfo);
+      };
+
+      this.selectAuthConfigId = function () {
+        var pluginInfo = _.find(PluginInfos(), function (pi) {
+          return pi.id() === args.authConfig().pluginId();
+        });
+
+        setPluginInfo(pluginInfo);
+        if (!pluginInfo) {
+          return;
+        }
+
+        PluginInfos.PluginInfo.get(pluginInfo.id()).then(function (pluginInfo) {
+          setPluginInfo(pluginInfo);
+          var newAuthConfig = new AuthConfigs.AuthConfig({id: args.authConfig().id(), pluginId: pluginInfo.id()});
+          newAuthConfig.etag(args.authConfig().etag());
+          args.authConfig(newAuthConfig);
+        });
+      };
+
+      this.clearMessage = function () {
+        args.message({});
+      };
+
+      this.getMessage = function () {
+        if (args.message && args.message() && args.message().type) {
+          return (
+            <f.row>
```

Comment:
is it possible to refactor this ? somehow returning "view (html part of the code)" from controller method looks odd to me.